corehq/util/metrics/metrics.py

```diff
@@ -86,7 +86,13 @@ def _create_event(self, title: str, text: str, alert_type: str = ALERT_INFO,
         pass
 
 
-Sample = namedtuple('Sample', ['type', 'name', 'tags', 'value'])
+class Sample(namedtuple('Sample', ['type', 'name', 'tags', 'value'])):
+    def match_tags(self, tags):
+        missing = object()
+        return all([
+            self.tags.get(tag, missing) == val
+            for tag, val in tags.items()
+        ])
```

Comment:
It wouldn't make any difference here because there are always a small number of tags, but useful to know about when iterating a lot of items -- If you drop the inner list comprehension, `all` will return `False` as soon as it gets the first `False` (and similarly `any` will return `True` on the first `True` it finds).
```
$ python -m timeit 'all([x for x in range(1000000)])'
10 loops, best of 3: 38.4 msec per loop
$ python -m timeit 'all(x for x in range(1000000))' 
1000000 loops, best of 3: 0.428 usec per loop
```